geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaProducerFeatureStore.scala

```diff
@@ -0,0 +1,116 @@
+package org.locationtech.geomesa.kafka
+
+import java.nio.charset.StandardCharsets
+import java.{util => ju}
+
+import com.vividsolutions.jts.geom.Envelope
+import kafka.producer.{KeyedMessage, Producer, ProducerConfig}
+import org.geotools.data.store.{ContentEntry, ContentFeatureStore}
+import org.geotools.data.{FeatureReader, FeatureWriter, Query}
+import org.geotools.feature.{FeatureReaderIterator, FeatureCollection}
+import org.geotools.feature.collection.{BridgeIterator, DelegateFeatureIterator}
+import org.geotools.filter.identity.FeatureIdImpl
+import org.geotools.geometry.jts.ReferencedEnvelope
+import org.geotools.referencing.crs.DefaultGeographicCRS
+import org.locationtech.geomesa.feature.{AvroFeatureEncoder, AvroSimpleFeature}
+import org.locationtech.geomesa.utils.text.ObjectPoolFactory
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.opengis.filter.identity.FeatureId
+import org.opengis.filter.{Filter, Id}
+
+import scala.collection.JavaConversions._
+
+object KafkaProducerFeatureStore {
+  val DELETE_KEY = "delete".getBytes(StandardCharsets.UTF_8)
+}
+
+class KafkaProducerFeatureStore(entry: ContentEntry,
+                                schema: SimpleFeatureType,
+                                broker: String,
+                                query: Query)
+  extends ContentFeatureStore(entry, query) {
+
+  val typeName = entry.getTypeName
+
+  override def getBoundsInternal(query: Query) =
+    ReferencedEnvelope.create(new Envelope(-180, 180, -90, 90), DefaultGeographicCRS.WGS84)
+
+  override def buildFeatureType(): SimpleFeatureType = schema
+
+  type FW = FeatureWriter[SimpleFeatureType, SimpleFeature]
+
+  val writerPool = ObjectPoolFactory(new ModifyingFeatureWriter(query), 5)
```

Comment:
Seemed to be just getting objects out of the pool. This is back when i was profiling our encoding/decoding. May not be a big issue depending on how expensive the function you're using it with is.